pkg/controllers/jobset_controller.go

```diff
@@ -154,6 +154,10 @@ func (r *JobSetReconciler) reconcile(ctx context.Context, js *jobset.JobSet, upd
 
 	// If JobSet is already completed or failed, clean up active child jobs and requeue if TTLSecondsAfterFinished is set.
 	if jobSetFinished(js) {
+		if err := r.deleteJobs(ctx, ownedJobs.active); err != nil {
```

Comment:
I think this change makes sense, although I think we originally put deleteJobs after the TTL check because users wanted to use TTL to allow time to inspect logs via kubectl before everything is deleted, to debug more easily without having to craft queries in cloud logging systems.

Quickly freeing up resources that may still be being used by a completed/failed JobSet is  important as well, but for this purpose the user can just tune their TTL accordingly. 

@CecileRobertMichon @kannon92 @ahg-g any thoughts on this?

Reply:
Sgtm.

Reply:
This is only deleting active jobs, which I think makes sense.

/lgtm
/approve